### DISCUSSION TITLE
获取群中用户列表获取到的id并不是微信号 无法查找到账号 添加好友失败！

### DISCUSSION THREAD

**Discussion by 360884617** [wechaty/wechaty#1934] [General]
🚨 The issue tracker is not for questions 🚨

If you have a question, please ask it on https://stackoverflow.com/questions/tagged/wechaty

[question]
获取群中用户列表获取到的id并不是微信号 无法查找到账号 添加好友失败！
![image](https://user-images.githubusercontent.com/63029163/78204796-22d93780-74cd-11ea-83ee-c403f59bf19f.png)
![image](https://user-images.githubusercontent.com/63029163/78204767-105efe00-74cd-11ea-9844-6cf0ddb212e3.png)
![image](https://user-images.githubusercontent.com/63029163/78204797-253b9180-74cd-11ea-8266-1866409bb892.png)

**Comment by su-chang** [comment]
Thanks for your issue.

This problem is due to the puppet which you used is not supported get room member's WeChat account, it could only get the `wxid` of room member who is not your friends.

So we could not add room member as friends by puppet `wechaty-puppet-padplus` or `wechaty-puppet-macpro`.

**Comment by 360884617** [comment]
@su-chang 我看官方文档写的支持，那我应该怎么处理，才可以添加群里的人呢

**Comment by su-chang** [comment]
Sorry, it is not supported by `wechaty-puppet-padplus` or `wechaty-puppet-macpro`.

**Comment by 360884617** [comment]
@su-chang 
项目有支持微信添加好友的功能吗 
![image](https://user-images.githubusercontent.com/63029163/78207448-49e73780-74d4-11ea-819b-da9e78da5097.png)

**Comment by su-chang** [comment]
If you can login in web wechat, [click me to test](https://wx.qq.com/).

You could have a try for [wechaty-puppet-puppeteer](https://github.com/wechaty/wechaty-puppet-puppeteer)

**Comment by su-chang** [comment]
> @su-chang
> 项目有支持微信添加好友的功能吗
> ![image](https://user-images.githubusercontent.com/63029163/78207448-49e73780-74d4-11ea-819b-da9e78da5097.png)

Of course, see : https://wechaty.js.org/api/friendship#friendship-add-contact-hello-promise-less-than-void-greater-than

**Comment by 360884617** [comment]
@su-chang wechat官方网页版的都提示不能登陆  
![image](https://user-images.githubusercontent.com/63029163/78207898-9717d900-74d5-11ea-87ff-9d5e01d4dec6.png)

**Comment by vaevas** [comment]
@su-chang 
![image](https://user-images.githubusercontent.com/51769860/78208262-86b42e00-74d6-11ea-92b1-7f428c42ef7a.png)
需要发送好友请求的联系人是通过什么获得呢?
还是固定已知的微信号实现添加好友呢?

**Comment by su-chang** [comment]
> wechat官方网页版的都提示不能登陆

It means that you can not use `wechaty-puppet-puppeteer`.

> 还是固定已知的微信号实现添加好友呢?

What you think is right.

**Comment by 360884617** [comment]
@su-chang  我必须知道别人的微信号 才可以吗 ？  
![image](https://user-images.githubusercontent.com/63029163/78208973-5b324300-74d8-11ea-924d-d49a32483a72.png)
这个通过这个ID是查找不到的    有没有别的处理方式  可以在群里 自动加好友

**Comment by su-chang** [comment]
Of course, just like what you can do in WeChat App. We could add friend only by phone number or WeChat account.

Sorry, add room member as friends is not support now.

**Comment by vaevas** [comment]
@su-chang 感谢您的回答 谢谢您

**Comment by su-chang** [comment]
@vaevas You are welcome.

**Comment by terranc** [comment]
看完上面的，到底有没有可能获取到群成员的微信号？
还是说仅仅pad和macos的puppet不行？那有哪个puppet可以的没？

**Comment by huan** [comment]
@terranc I believe the WXWork, Donut, Rock, and PadLocal are all supporting to get the wxid of the room members.

Learn more about the Wechaty Puppet Services at https://wechaty.js.org/docs/puppet-services/
